app/services/whatsapp/incoming_message_service_helpers.rb

```diff
@@ -104,4 +104,16 @@ def find_message_by_source_id(source_id)
 
     @message = Message.find_by(source_id: source_id)
   end
+
+  def message_under_process?
+    Redis::Alfred.get(@processed_params[:messages].first[:id])
+  end
+
+  def set_message_source_id
+    ::Redis::Alfred.setex(@processed_params[:messages].first[:id], true) if @processed_params.try(:[], :messages).present?
```

Comment:
Lets use a namespace Redis key here : 
Example: https://github.com/chatwoot/chatwoot/blob/847d7ea082ac6761d3949fd6209f885c5925acf5/lib/redis/redis_keys.rb#L10

will help us to group / target specific keys related to this fix alone in future.

Reply:
@sojan-official Done with the changes. Please have a look.